### PR TITLE
GEODE-7563: eliminate home-grown ConcurrentHashSet

### PR DESCRIPTION
**Pull request by Bill**
[GEODE-7563](https://issues.apache.org/jira/browse/GEODE-7563)

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/DiskInitFile.java [hunk 1]
```diff
@@ -336,9 +335,7 @@
 
   private final LongOpenHashSet crfIds;
```
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/internal/cache/DiskInitFile.java]: I realize these variable names are pre-existing... any chance to make them human understandable...?

Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/internal/cache/DiskInitFile.java]: I'm calling that out of scope.

**Comment by Bill** [comment]
Since I had to remove the `ConcurrentHashSet` entry from `geode-core/src/main/resources/org/apache/geode/internal/sanctioned-geode-core-serializables.txt` the next thing I have to do is analyze the places where `ConcurrentHashSet` was used (Java classes) to see which, if any, were actually serialized (and which, therefore, might break as a result of the changes in this PR)

File: geode-core/src/main/java/org/apache/geode/internal/cache/DiskInitFile.java (hunk 1 above)
Comment (jhuynh1) [reply geode-core/src/main/java/org/apache/geode/internal/cache/DiskInitFile.java]: Also, i think crf Ids is the readable form in this case :-)

**Pull request merged**